lineax/_solver/tridiagonal.py

```diff
@@ -50,51 +54,29 @@ def init(self, operator: AbstractLinearOperator, options: dict[str, Any]):
             )
         return tridiagonal(operator), pack_structures(operator)
 
+    @classmethod
+    def abstract_eval(
+        cls, diagonal, lower_diagonal, upper_diagonal, vector, batch_unroll_gpu
+    ):
+        return ShapedArray(vector.shape, vector.dtype)
```

Comment:
I agree with your comment, this should be a global function!

As for its generality, I think it's basically fine -- we perform the solve on the ravelled form of the input, so just a single array is what's expected.